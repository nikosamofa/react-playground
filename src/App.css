.App{
display: flex;
margin: 40px;
min-height: calc(100vh - 80px)
}

.left {
    background-color: darkslategray;
    color: azure;
  }
  
  .right {
    background-color: wheat;
    color: firebrick;
  }
  div {
    margin: 10px;
    display: flex;
    flex-direction: column;
  }