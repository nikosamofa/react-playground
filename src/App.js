// import React from 'react';
// import Split from './composition/Split';
// import './App.css';
// import Tooltip from './composition/Tooltip';


// const firstTooltip = (
//   <Tooltip message='tooltip message' color='hotpink'>
//     ipsum 
//   </Tooltip>
// )
// const secondTooltip=(
//   <Tooltip message='another tooltip message' color='#126bcc'>
//     officiis
//   </Tooltip>
// )

// function App() {
//   return (
//     <main className='App'>
//       <Split className='left' flexBasis='2'>
//         This is the content for the left Split. Lorem {firstTooltip} dolor sit amet consectetur, adipisicing elit. Incidunt ex velit suscipit facere officia?<br />
//         {/* make another tooltip directly inside the App */}
//         <Tooltip message='one more tooltip message'>
//           Necessitatibus?
//         </Tooltip>
//       </Split>
//       <Split className='right'>
//         This is the content for the right Split. Inventore aliquid cupiditate suscipit repellat. Quaerat quis {secondTooltip} quam fuga. Aliquid quo possimus id soluta aspernatur.
//       </Split>
//     </main>
//   )
// }
// export default App;
import React, { Component } from 'react';
import './App.css';
import Messages from './Messages';
import TheDate from './state/TheDate';
import Counter from './state/Counter'
import HelloWorld from './state-drills/HelloWorld';
import Bomb from './state-drills/Bomb'
import RouletteGun from './state-drills/RouletteGun'
import Tabs from './state/Tabs'
import Accordion from './state-drills/Accordion'

const tabsProp = [
  {
    name: 'First tab',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.'
  },
  {
    name: 'Second tab',
    content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.'
  },
  {
    name: 'Third tab',
    content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.'
  },
];
const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]
class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>YOUR APPLICATION NAME!</h1>
        <div>Hello!</div>
        <Messages name='Messages' unread={0} />
        <Messages name='Notifications' unread={10} />
        <TheDate />
        <Counter count={123} />
        <HelloWorld />
        <Bomb />
        <RouletteGun />
        <Tabs tabs={tabsProp} />
        <Accordion sections={sections} />
      </div>
    );
  }
}

export default App;


