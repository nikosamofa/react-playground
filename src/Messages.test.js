import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Messages from './Messages';
import renderer from 'react-test-renderer';

describe('Message component', ()=>{
//this is the smoke test
it('should render without crashing', ()=>{
    const div = document.createElement('div');

    ReactDOM.render(<Messages />, div);

    ReactDOM.unmountComponentAtNode(div)
}
)

//snapshot renderer

it('renders the UI as expected', ()=>{
    const tree = renderer.create(<Messages name='Messages' unread={4} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

//test for when unread is 0
it('renders the UI without crashing', ()=>{
    const tree = renderer.create(<Messages name='Messages' unread={0}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

})
