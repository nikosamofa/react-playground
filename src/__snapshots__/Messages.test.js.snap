// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Message component renders the UI without crashing 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`Message tests renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Message tests renders the UI without crashing 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count"
  >
    4
  </div>
</div>
`;

exports[`renders the UI without crashing 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
